src/test/testUtils.tsx

```diff
@@ -60,7 +63,46 @@ export const filterLoggingInTests = (
   });
 };
 
+export async function loadFixtureData(testName: string): Promise<UIStore> {
+  const fixtureData = JSON.parse(
+    readFileSync(
+      join(__dirname, "..", "..", "public", "test", "fixtures", `${testName}.js.json`),
+      "utf8"
+    )
+  );
+
+  const sessionId = fixtureData.find((message: any) => {
+    if (message.hasOwnProperty("sessionId")) {
+      return message.sessionId;
+    }
+  });
+
+  if (!sessionId) {
+    console.warn("Fixture does not contain a session ID");
+  }
+
+  // TODO This is side effectful in a way that affects ThreadFront.setSessionId(); we should clean that up!
+  const store = await createTestStore();
+
+  // This is necessary to unblock various event listeners and parsing.
+  // Actual session ID value _probably_ doesn't matter here.
+  await ThreadFront.setSessionId(sessionId);
+
+  // Initialize state using exported websocket messages,
+  // sent through the mock environment straight to socket parsing.
+  fixtureData.forEach((message: any) => {
+    window.mockEnvironment?.sendSocketMessage(JSON.stringify(message));
+  });
+
+  // Give everything time to settle
+  await new Promise(resolve => setTimeout(resolve, 100));
+
+  return store;
+}
+
 export const filterCommonTestWarnings = () => {
+  // Skip LoadedRegions debug messages
+  filterLoggingInTests(message => message === "LoadedRegions", "debug");
```

Comment:
These debug logs are kind of annoying so I added this (for now). Maybe we should always filter out _all_ `console.debug` logs from test output.